Providers/Profile.swift

```diff
@@ -1082,8 +1082,29 @@ open class BrowserProfile: Profile {
                                   account: FirefoxAccount,
                                   statsSession: SyncOperationStatsSession, why: SyncReason) -> Deferred<Maybe<[(EngineIdentifier, SyncStatus)]>> {
             log.info("Syncing \(synchronizers.map { $0.0 })")
-            let authState = account.syncAuthState
+            var authState = account.syncAuthState
             let delegate = self.profile.getSyncDelegate()
+
+            var enginesEnablements: [String: Bool] = [:]
+            // We just created the account, the user went through the Choose What to Sync screen on FxA.
+            if let declined = account.declinedEngines {
```

Comment:
Can we break this whole block of code out into a named method: `func engineEnablementChangesForAccount(account: Account, profile: Profile) -> [String: Bool]?`